core/src/main/resources/jenkins/views/JenkinsHeader/headerContent.jelly

```diff
@@ -0,0 +1,80 @@
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:i="jelly:fmt" xmlns:x="jelly:xml">
+	<header id="header" class="page-header">
+        <div class="page-header__brand">
+            <div class="logo">
+                <a id="jenkins-home-link" href="${rootURL}/">
+                    <img id="jenkins-head-icon" src="${imagesURL}/jenkins-header-logo-v2.svg" alt="[${logoAlt}]" />
+                    <img id="jenkins-name-icon" src="${imagesURL}/title.svg" alt="${title}" width="139" height="34" />
+                </a>
+            </div>
+
+            <a class="page-header__brand-link" href="${rootURL}/">
+                <img src="${imagesURL}/jenkins-header-logo-v2.svg"
+                     alt="[${logoAlt}]"
+                     class="page-header__brand-image" />
+                <span class="page-header__brand-name">Jenkins</span>
+            </a>
+        </div>
+
+        <div class="searchbox hidden-xs">
```

Comment:
biggest concern with this PR is about making this header an API, what happens when we want to change it and it breaks plugins that have customised it.

Could you show a reference implementation that shows a realistic scenario and the changes that were needed to be done.

i.e. would it be better to create specific APIs along the lines of:

* Logo
* Area to the right of the logo
* ...

Reply:
A reference implementation is included in the JEP description. See [JEP-234](https://github.com/imonteroperez/jep/blob/jep-401-header/jep/234/README.adoc)

Reply:
See: https://github.com/imonteroperez/custom-header-plugin/tree/main/src/main

Reply:
Agreed that this is the main concern. https://github.com/jenkinsci/jep/pull/380#discussion_r749337871 Is the current design suitable only for headers which totally replace the stock GUI with a completely different style? Seems to be impossible to incrementally modify particular pieces. Possibly OK if so, but needs to be emphasized in the JEP and **Reasoning** explaining this choice.

Reply:
JEP updated on reasoning section https://github.com/jenkinsci/jep/pull/380#commits-pushed-2499ba3